src/ba.py

```diff
@@ -11,15 +11,29 @@ def border_array(x: str) -> list[int]:
     [0, 0, 0, 1, 2, 3, 4, 0, 0, 1]
     >>> border_array("")
     []
+    >>> strict_border_array("abaabaa")
+    [0, 0, 1, 1, 2, 3, 4]
     """
-    return []  # FIXME
+    if x=="":
+        return []
+    border_list = [0]
+    index_match = 0
+    for c in x[1:]:
+        if c==x[index_match]:
+            index_match += 1
+        elif c==x[0]:
+            index_match = 1
+        else:
+            index_match = 0
```

Comment:
This code is essentially saying that if you can extend the previous border you do that (correct) and if not, the longest border is either the first character in the string or empty (not correct).

Consider the string `abcabdabcabc`. (You might be able to find a shorter example, but this is what I could think of right now).

You will correctly identify that the first three values should be zero, since there are no non-empty borders there. Then you identify that the next three positions are `abd` where the first has a border of length 1, the second of length 2, and the third of length 0. So, the border array is now `[0, 0, 0, 1, 2, 0, ...]`. Again, everything is great so far.

After the reset when we see the `d`, we build up `abcab` again and get the numbers `[..., 1, 2, 3, 4, 5, ...]` because we are matching `a`, `ab`, abc`, `abca` and `abcab`. So far, everything is still going well. But then we see `c`.

We cannot extend the `c` (`abcab_d_ != abcab_c_`), so we have to look at a shorter border. You jump down to index zero and see that `a != c` and conclude that the longest border here has length zero. This, of course, isn't true. The string `abc` is a border. The right answer should be three.

Your problem is that you do not consider the previous string's borders in decreasing length. If you can't get the perfect situation--where you can extend the previous border--you assume the worst.